src/main/java/bgu/spl/mics/application/passiveObjects/Diary.java

```diff
@@ -38,23 +63,33 @@ public void addReport(Report reportToAdd){
 	 * List of all the reports in the diary.
 	 * This method is called by the main method in order to generate the output.
 	 */
-	public void printToFile(String filename){
-		//TODO: Implement this
+	public void printToFile(String filename) throws IOException {
+		_lock.readLock().lock();
+		Gson gson = new Gson();
+		String reportsToPrint = gson.toJson(_diary);
+		PrintWriter writer = null;
```

Comment:
?

Reply:
Considering I straight up stole it from your Inventory implementation you should probably upload an update to it :)